tests/adapter/dbt/tests/adapter/constraints/fixtures.py

```diff
@@ -116,4 +130,21 @@
         data_type: text
       - name: date_day
         data_type: date
+  - name: my_model_wrong_data_type
+    config:
+      contract: true
+    columns:
+      - name: id
+        data_type: integer
+        description: hello
+        constraints: ['not null','primary key']
+        constraints_check: (id > 0)
+        tests:
+          - unique
+      - name: color
+        data_type: text
+      - name: date_day
+        data_type: date
+      - name: num_array
+        data_type: int[]
```

Comment:
Would there be value in expanding this to cover more possible values for `data_type`? e.g., `boolean`, `timestamp without time zone`, `timestamp with time zone`, etc?

Or would that type of coverage be better in some other test?

Reply:
Definitely - I'm working on rejigging the tests to have more coverage here + be easier to test adapter-specific types.

Reply:
One idea for testing using adapter-specific values is to use the data type functions [here](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#:~:text=type_timestamp-,Data%20type%20functions,-type_bigint).

<img width="175" alt="image" src="https://user-images.githubusercontent.com/44704949/222602913-d4958fb9-1581-4674-84b2-ab03e2be95e1.png">

e.g., something like:
```
      - name: color
        data_type: {{ dbt.type_string() }}
```

One downside of this idea is that it doesn't cover all the data types you might expect. For starters, it doesn't include `date` or `time`. Also, most databases support at least two different timestamp data types, but `type_timestamp()` only covers one of those.